src/app/shared/service/mind-share.service.ts

```diff
@@ -24,12 +26,15 @@ export class MindShareService {
   }
 
   getMindShare(id: string): Observable<MindShare> {
-    return this.getMindShares().map(mindShares => {
-      const result = mindShares.filter(mindShare => mindShare.id === id)[0];
-      if (result) {
-        result.publishDate = new Date(result.publishDate);
-      }
-      return result;
+    return this.httpClient.get<MindShare>(this.url + '/' + id).map(mindShare => {
+      mindShare.publishDate = new Date(mindShare.publishDate);
+      return mindShare;
     });
   }
+
+    getFeaturedMindShare(): Observable<MindShare> {
+      return this.getMindShares().map(mindShares => {
+        return mindShares.filter(mindShare => mindShare.isFeatured === true)[0];
```

Comment:
We can clean this up to just say:

`return mindShares.filter(mindShare => mindShare.isFeatured)[0];`